subiquity/server/controllers/filesystem.py

```diff
@@ -1200,6 +1200,60 @@ def available_use_gap_scenarios(
             scenarios.append((gap.size, use_gap))
         return scenarios
 
+    def resize_has_enough_room_for_partitions(self, disk, resized: Partition) -> bool:
+        """Check if we have enough room for all the primary partitions. This
+        isn't failproof but should limit the number of TargetResize scenarios
+        that are suggested but can't be applied because we don't have enough
+        room for partitions."""
+        new_primary_parts = 0
```

Comment:
A common util function between this and use_gap would be nice.

Reply:
Refactored! Thanks